plugins/callback/mail.py

```diff
@@ -103,18 +122,22 @@ def mail(self, subject='Ansible error mail', body=None):
 
         content = 'Date: %s\n' % email.utils.formatdate()
         content += 'From: %s\n' % self.sender
-        content += 'To: %s\n' % self.to
+        if self.to:
+            content += 'To: %s\n' % ','.join(self.to)
         if self.cc:
-            content += 'Cc: %s\n' % self.cc
+            content += 'Cc: %s\n' % ','.join(self.cc)
         content += 'Message-ID: %s\n' % email.utils.make_msgid()
         content += 'Subject: %s\n\n' % subject.strip()
         content += body
 
-        addresses = self.to.split(',')
+        addresses = self.to
         if self.cc:
-            addresses += self.cc.split(',')
+            addresses += self.cc
         if self.bcc:
-            addresses += self.bcc.split(',')
+            addresses += self.bcc
+
+        if not addresses:
+            self._display.warning('No receiver has been specified for the mail callback plugin')
```

Comment:
```suggestion
            self._display.warning('No receiver has been specified for the mail callback plugin.')
```
(if you did it intentionally, just mark as resolved)